crates/rover-client/src/error.rs

```diff
@@ -20,12 +20,34 @@ pub enum RoverClientError {
     InvalidJSON(#[from] serde_json::Error),
 
     /// Encountered an error handling the received response.
-    #[error("encountered an error handling the response: {msg}")]
-    HandleResponse {
+    #[error("{msg}")]
+    AdhocError {
         /// The error message.
         msg: String,
     },
 
+    /// The user provided an invalid subgraph name.
+    #[error("Could not find subgraph \"{invalid_subgraph}\".")]
+    NoSubgraphInGraph {
+        /// The invalid subgraph name
+        invalid_subgraph: String,
+
+        /// A list of valid subgraph names
```

Comment:
```suggestion
        /// A list of valid subgraph names which can be used by other error libraries to provide a suggestion
```

This just took me a second to figure out that the valid subgraphs is used by rover-error